name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/MeasurementTool.java

```diff
@@ -91,22 +94,34 @@ public void setColor(Color color)
 
     public void addButtons(ToolBarManager toolBar)
     {
-        toolBar.add(createAction());
+        var action = createAction(false);
+        // store buttons to update their image on context menu action
+        buttons.add(action);
+        toolBar.add(action);
     }
 
     public void addContextMenu(IMenuManager manager)
     {
-        manager.add(createAction());
+        manager.add(createAction(true));
     }
 
-    private SimpleAction createAction()
+    private SimpleAction createAction(boolean doUpdateButton)
     {
         return new SimpleAction(Messages.LabelMeasureDistance, //
                         isActive ? Images.MEASUREMENT_ON : Images.MEASUREMENT_OFF, //
                         a -> {
                             isActive = !isActive;
-                            a.setImageDescriptor(isActive ? Images.MEASUREMENT_ON.descriptor()
-                                            : Images.MEASUREMENT_OFF.descriptor());
+                            ImageDescriptor image = isActive ? Images.MEASUREMENT_ON.descriptor()
+                                            : Images.MEASUREMENT_OFF.descriptor();
+
+                            // update image of button
+                            if (doUpdateButton)
+                                buttons.forEach(b -> {
+                                    b.setImageDescriptor(image);
+                                    b.setChecked(isActive);
```

Comment:
This ```isChecked``` has no effect unless we set the style bit ```IAction.AS_CHECK_BOX``` on the action.

If we do this, then it looks like this:

<img width="119" alt="Bildschirmfoto 2023-04-18 um 18 23 22" src="https://user-images.githubusercontent.com/587976/232842015-32151f88-2b18-4480-a6cf-a38833d874f5.png">

I do not like that form the UI/UX design (and the only reason I have enabled it for the time periods "1Y", "2Y" etc. is that it is not possible to change the text color).

However, what I do not know if there is any benefit for screen readers. Do you know?

For now, I will remove the setChecked and merge the proposal. LGTM

Reply:
> This `isChecked` has no effect unless we set the style bit `IAction.AS_CHECK_BOX` on the action.

Oh, I have not consiered that, thank you for the clarification, @buchen.
I don't know about if screen reader can handle this.

By the way, I was thinking of looking for another icon. I think it looks like a hair comb. Because the lines of the ruler look like gaps.